flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -504,23 +502,18 @@ protected void beforeInvoke() throws Exception {
 
 	private void readRecoveredChannelState() throws IOException, InterruptedException {
 		SequentialChannelStateReader reader = getEnvironment().getTaskStateManager().getSequentialChannelStateReader();
-		if (reader.hasChannelStates()) {
-			reader.readOutputData(getEnvironment().getAllWriters());
-			channelIOExecutor.execute(() -> {
-				try {
-					reader.readInputData(getEnvironment().getAllInputGates());
-				} catch (Exception e) {
-					asyncExceptionHandler.handleAsyncException("Unable to read channel state", e);
-				}
-			});
-
-			for (InputGate inputGate : getEnvironment().getAllInputGates()) {
-				inputGate
-					.getStateConsumedFuture()
-					.thenRun(() -> mainMailboxExecutor.execute(inputGate::requestPartitions, "Input gate request partitions"));
+		reader.readOutputData(getEnvironment().getAllWriters());
+		channelIOExecutor.execute(() -> {
+			try {
+				reader.readInputData(getEnvironment().getAllInputGates());
+			} catch (Exception e) {
+				asyncExceptionHandler.handleAsyncException("Unable to read channel state", e);
 			}
-		} else {
-			requestPartitions();
+		});
+		for (InputGate inputGate : getEnvironment().getAllInputGates()) {
```

Comment:
Okay you just did what I wrote on the last commit. Probably squashing makes sense.